src/main/java/com/systemmeltdown/robotlib/subsystems/drive/SingleSpeedTalonDriveSubsystemWithTrajectorySubsystem.java

```diff
@@ -0,0 +1,216 @@
+            m_rightTalonGroup = rightTalonGroup;
+            m_leftTalonGroup = leftTalonGroup;
+            m_leftEncoder.setDistancePerPulse(Constants.ENCODER_DISTANCE_PER_PULSE);
+            m_rightEncoder.setDistancePerPulse(Constants.ENCODER_DISTANCE_PER_PULSE);
+        
+            resetEncoders();
+            m_gyro = new PigeonIMU(gyroID);
+            m_gyro.configFactoryDefault();
+            m_odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getHeading()));
+    }
+
+    @Override
+    public void periodic() {
+      // Update the odometry in the periodic block
+      m_odometry.update(Rotation2d.fromDegrees(getHeading()), m_leftEncoder.getDistance(),
+                        m_rightEncoder.getDistance());
+    }
+
+    @Override
+    protected int getCurrentSpeedLeftClicksPerSecond() {
+        return m_leftTalonGroup.getSelectedSensorPosition();
+    }
+
+    @Override
+    protected int getCurrentSpeedRightClicksPerSecond() {
+        return m_rightTalonGroup.getSelectedSensorPosition();
+    }
+
+    @Override
+    protected void setProportional(double leftProportion, double rightProportion) {
+        m_leftTalonGroup.set(ControlMode.PercentOutput, leftProportion);
+        m_rightTalonGroup.set(ControlMode.PercentOutput, rightProportion);
+    }
+
+    @Override
+    public void configure(Map<String, Object> config) {
+        super.configure(config);
+        m_leftTalonGroup.configure(((Boolean) config.get(CONFIG_IS_LEFT_INVERTED)).booleanValue());
+        m_rightTalonGroup.configure(((Boolean) config.get(CONFIG_IS_RIGHT_INVERTED)).booleanValue());
+    }
+
+    @Override
+    protected void setVelocity(int leftClicksPerSecond, double rightClicksPerSecond) {
+        // TODO implement PID Loop
+    }
+
+    /**
+     * Returns the currently-estimated pose of the robot.
+     *
+     * @return The pose.
+     */
+    public Pose2d getPose() {
+        return m_odometry.getPoseMeters();
+      }
+  
+    /**
+     * Resets the odometry to the specified pose.
+     *
+     * @param pose The pose to which to set the odometry.
+     */
+    public void resetOdometry(Pose2d pose) {
+        resetEncoders();
+        m_odometry.resetPosition(pose, Rotation2d.fromDegrees(getHeading()));
+    }
+
+    /**
+     * Returns the current wheel speeds of the robot.
+     *
+     * @return The current wheel speeds.
+     */
+    public DifferentialDriveWheelSpeeds getWheelSpeeds() {
+        return new DifferentialDriveWheelSpeeds(m_leftEncoder.getRate(), m_rightEncoder.getRate());
+      }
+    
+    /**
+     * Controls the left and right sides of the drive directly with voltages.
+     *
+     * @param leftVolts  the commanded left output
+     * @param rightVolts the commanded right output
+     */
+    public void setTankDriveVolts(double leftVolts, double rightVolts) {
+        m_rightTalonGroup.setMasterTalonVolts(leftVolts);
+        m_leftTalonGroup.setMasterTalonVolts(-rightVolts);
```

Comment:
If these are supposed to be reversed, add a comment explaining why. It looks like a mistake.